File: pulse-go/internal/pulse/uci/engine.go
/*
 * Copyright 2013-2023 Phokham Nonava
 *
 * Use of this source code is governed by the MIT license that can be
 * found in the LICENSE file.
 */

package uci

import "github.com/fluxroot/pulse/internal/pulse/engine"

//go:generate mockgen -source=engine.go -destination=mock/engine.go -package=mock

type Engine interface {
	Initialize() error
	Ready() error
	SetNameOnlyOption(name string)
	SetNameValueOption(name string, value string)
	NewGame() error
	Position(p *engine.Position)
	Start() error
	Stop() error
	PonderHit() error
	Quit() error
}

File: pulse-kotlin/src/commonMain/kotlin/com/fluxchess/pulse/kotlin/uci/Engine.kt
/*
 * Copyright 2013-2024 Phokham Nonava
 *
 * Use of this source code is governed by the MIT license that can be
 * found in the LICENSE file.
 */

package com.fluxchess.pulse.kotlin.uci

import com.fluxchess.pulse.kotlin.engine.Position

interface Engine {
	fun initialize()
	fun ready()
	fun setOption(name: String)
	fun setOption(name: String, value: String)
	fun newGame()
	fun position(position: Position)
	fun start()
	fun stop()
	fun ponderHit()
	fun quit()
}
